atd-kits/signal_status_publisher.py

```diff
@@ -90,6 +99,31 @@ def merge_signal_asset_data(kits_sig_status, signal_asset_data):
         # stop this script. likely an issue with dupes in the signal assets ETL
         matched_signal = matched_signal_list[0]
         kits_signal.update({key: matched_signal.get(key) for key in asset_fields})
+        if kits_signal["signal_id"] in dark_signal_ids:
+            kits_signal["operation_state"] = 4
+            kits_signal["operation_text"] = "Dark Traffic Signal (No Power)"
+            # overwriting operation datetime with the date the knack record was last updated
+            kits_signal["operation_state_datetime"] = arrow.get(
+                matched_signal["modified_date"][:-5], "YYYY-MM-DDTHH:mm:ss"
+            ).datetime
+
+    # Catching dark signals not captured in the initial KITS query
+    signal_ids = [signal["signal_id"] for signal in kits_sig_status]
+    for dark_signal in dark_signal_data:
+        if dark_signal["signal_id"] not in signal_ids:
+            # Building dark signal record
+            rec = {
+                "signal_id": dark_signal["signal_id"],
+                "operation_state": 4,
+                "operation_text": "Dark Traffic Signal (No Power)",
+                "plan_id": -1,
+                # operation datetime is the date the knack record was last updated
+                "operation_state_datetime": arrow.get(
+                    dark_signal["modified_date"][:-5], "YYYY-MM-DDTHH:mm:ss"
+                ).datetime,
+            }
+            rec.update({key: dark_signal.get(key) for key in asset_fields})
+            kits_sig_status.append(rec)
```

Comment:
these two blocks of code for creating a dark signal seem redundant. could just have one function that does this?